django/test/signals.py

```diff
@@ -178,3 +178,10 @@ def static_finders_changed(**kwargs):
     }:
         from django.contrib.staticfiles.finders import get_finder
         get_finder.cache_clear()
+
+
+@receiver(setting_changed)
+def auth_user_model_changed(**kwargs):
+    if kwargs['setting'] == 'AUTH_USER_MODEL':
+        from django.apps import apps
+        apps.get_swappable_setting.cache_clear()
```

Comment:
Let me put this differently :-) Is this required to make the test suite pass? If not, I'd prefer we do not include it. If yes, I'd like to look at the failing tests, because they must be weird.